### PR TITLE
Add hard edge clip behavior to Material link bottom sheet modal

### LINKED ISSUE #331: Link menu lacks bottom padding and clipping
## Steps to Reproduce

1. Insert a link
2. Open the link menu
3. Use the buttons

The button at the bottom doesn't take into account the device controls and displays under them, some padding should be added.

The button at the top isn't clipped and displays outside the bottom sheet (when selected).

https://github.com/fleather-editor/fleather/assets/44336112/46f79758-cbb3-47a1-8458-60536a3c587f

## Environment
OS: Android
Flutter version: 3.19.3
Fleather version: 1.14.4

### PR DESCRIPTION
**Pull request by amantoux**
Fixes #331

### CODE REVIEW

**Review by Amir-P — COMMENTED** [review]
Happy to see you back again!
I think the OP had an issue with safe are and bottom padding too. Could you please fix that too?

**Comment by amantoux** [comment]
> Happy to see you back again! I think the OP had an issue with safe are and bottom padding too. Could you please fix that too?

Not sure why the padding would be needed
I added `useSafeArea`. (Funny the default is `false`)

File: packages/fleather/lib/src/widgets/link.dart [hunk 1]
```diff
@@ -109,6 +110,8 @@ class _CupertinoAction extends StatelessWidget {
 Future<LinkMenuAction> _showMaterialMenu(
     BuildContext context, String link) async {
   final result = await showModalBottomSheet<LinkMenuAction>(
+    clipBehavior: Clip.hardEdge,
+    useSafeArea: true,
```
Comment (Amir-P) [inline packages/fleather/lib/src/widgets/link.dart:114]: As it's been pointed out in https://github.com/fleather-editor/fleather/issues/331#issuecomment-2088701131, `useSafeArea` does not fix the issue. I tried wrapping the widget inside `builder` with `SafeArea` and seems to be working fine. Could you check it?

**Review by Amir-P — CHANGES_REQUESTED** [review]
PTAL.

File: packages/fleather/lib/src/widgets/link.dart (hunk 1 above)
Comment (amantoux) [reply packages/fleather/lib/src/widgets/link.dart]: As discussed, @Amir-P noticed the `useSafeArea` argument does not include padding with respect to the bottom.

**Review by Amir-P — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**